### PR TITLE
Bloc313 tags api get comments

### PR DESCRIPTION
**Pull request by alexplevako**

### CODE REVIEW

File: tests/chain_tests/plugins/tags/tags_common.hpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            operation.parent_author = my.author;
+            operation.parent_permlink = my.permlink;
+            c(operation);
+
+            if (operation.permlink.empty())
+                operation.permlink = tags_fixture::title_to_permlink(operation.title);
+
+            fixture->push_operation<comment_operation>(operation, actor.private_key);
+
+            fixture->generate_blocks(20 / SCORUM_BLOCK_INTERVAL);
```
Comment (Andrei-Masilevich) [inline tests/chain_tests/plugins/tags/tags_common.hpp]: Why generate_blocks()? push_operation has put_in_block = true already

File: tests/chain_tests/plugins/tags/tags_tests.cpp [hunk 2]
```diff
@@ -184,6 +72,139 @@ SCORUM_TEST_CASE(test_depth)
     BOOST_REQUIRE_EQUAL(2u, check_list[root_child_child.permlink()]);
 }
 
+SCORUM_TEST_CASE(get_discussions_by_created_return_post_and_it_comments_if_its_id_0)
+{
```
Comment (Andrei-Masilevich) [inline tests/chain_tests/plugins/tags/tags_tests.cpp]: Comment please this wrong behavior

File: libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp [hunk 3]
```diff
@@ -1043,6 +1067,70 @@ class tags_api_impl
         }
         return result;
     }
+
+    template <typename Index> class index_traverse
```
Comment (abolinbsu) [inline libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp]: Do you really need make this class as a template?

File: libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp [hunk 4]
```diff
@@ -1043,6 +1067,70 @@ class tags_api_impl
...
+                scan_children(itr->author, fc::to_string(itr->permlink));
+            }
+        }
+
+    private:
+        const Index& _index;
+
+        void scan_children(const account_name_type& parent_author, const std::string& parent_permlink)
+        {
+            auto itr = _index.find(boost::make_tuple(parent_author, parent_permlink));
```
Comment (abolinbsu) [inline libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp]: You can implement scan_children in more shorter way:
auto range = get_range_by<by_parent>(boost::make_tuple(parent_author, parent_permlink, 0) <= ::boost::lambda::_1,
                                             ::boost::lambda::_1 <= std::make_tuple(parent_author, parent_permlink, ALL_IDS));
std::reverse_copy (range.begin(), range.end(), _stack.begin());

Comment (Andrei-Masilevich) [reply libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp]: Good idea! It totally removes put_in_stack and change value type for std::stack from search_iterator to std::reference_wrapper<const object_type>. Look on_item(*itr). 
Just add method get_children to comment_service_i

**Pull request merged**